CMakeLists.txt

```
@@ -55,30 +73,87 @@ set(SRC_FILES
     src/egm_logger.cpp
     src/egm_udp_server.cpp
     src/egm_trajectory_interface.cpp
-    ${EgmProtoSources})
-
-include_directories(include
-                    ${CMAKE_CURRENT_BINARY_DIR} # Contains protobuf generated sources
-                    ${Boost_INCLUDE_DIRS}
-                    ${PROTOBUF_INCLUDE_DIRS}
-                    ${catkin_INCLUDE_DIRS})
+    ${EgmProtoSources}
+)
 
 add_library(${PROJECT_NAME} ${SRC_FILES})
-target_include_directories(${PROJECT_NAME} PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
-target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES} ${PROTOBUF_LIBRARIES} ${catkin_LIBRARIES})
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
```

Suggestion: let's be a little more creative with the namespace and exported targets:

```suggestion
add_library(ABB::libegm ALIAS ${PROJECT_NAME})
```

---

We'd need to something like this as well to do this though:

```cmake
set_target_properties(${PROJECT_NAME} PROPERTIES EXPORT_NAME libegm)
```

---

I do not feel strongly about this, but most CMake package that I know and almost all `Find***.cmake` scripts in CMake, have the namespace that corresponds to the package name, i.e. if you have `find_package(<package>)`, then your imported targets are `<package>::lib1`, `<package>::lib2` . I don't think this is documented or recommended in any CMake docs, but I feel this would go against the  " principle of least astonishment" , https://en.wikipedia.org/wiki/Principle_of_least_astonishment .

---

I like @traversaro's argument, but @gavanderhoorn's approach would "harmonize" better with e.g. [abb_librws](https://github.com/ros-industrial/abb_librws).

So, I am a little bit split about which to use.

---

@traversaro: least astonishment is certainly something to keep in mind. However, I always have to lookup what names a `Find*.cmake` defines, as even for 'old style' CMake, there are only conventions and quite a few libraries (even those for which find scripts are distributed with the 'official' CMake distribution) violate all of them (lower case, all upper case, snake-case, with or without dashes, etc, etc).

Proper documentation should be added in all cases I believe.

Capitalising the namespace was something I saw in [pabloariasal/modern-cmake-sample](https://github.com/pabloariasal/modern-cmake-sample/blob/cfda7b51ff915409fd61d4f2e75b63032bedb182/libjsonutils/CMakeLists.txt#L20) (from [this article](https://pabloariasal.github.io/2018/02/19/its-time-to-do-cmake-right)). Now that is `n==1` of course, so hardly convincing from that perspective, but I must say I like it.

Another example of where this is used is Boost. The main namespace is `Boost`, so we get `Boost::system`. Not `Boost_system::Boost_system`.

Having written that: it's just a name. If @jontje prefers least-astonishment, let's go with `abb_libegm::abb_libegm`.

---

> Capitalising the namespace was something I saw in [pabloariasal/modern-cmake-sample](https://github.com/pabloariasal/modern-cmake-sample/blob/cfda7b51ff915409fd61d4f2e75b63032bedb182/libjsonutils/CMakeLists.txt#L20) (from [this article](https://pabloariasal.github.io/2018/02/19/its-time-to-do-cmake-right)). Now that is `n==1` of course, so hardly convincing from that perspective, but I must say I like it.
> 
> Another example of where this is used is Boost. The main namespace is `Boost`, so we get `Boost::system`. Not `Boost_system::Boost_system`.

I am probably missing something, but as far as I can see, both example respect the convention that a package found by `find_package(<package>)` always exports CMake targets in the form of `<package>::<libname>` , in particular: 

| `<package>` |             `<libname>`            |  `<package>::<libname>` | 
|:-------------:|:-----------------:|:-------------------:|
| JSONUtils    | jsonutils            |  JSONUtils::jsonutils  |
| Boost          |  system             | Boost::system |

---

> Having written that: it's just a name. If @jontje prefers least-astonishment, let's go with `abb_libegm::abb_libegm`.

Same for me, bikeshedding is fun but I would feel fine with either decision.

---

Let's go with `abb_libegm::abb_libegm`.